filebeat/input/log/harvester.go

```diff
@@ -461,10 +461,7 @@ func (h *Harvester) onMessage(
 				"_id": id,
 			}
 		}
-	} else if &text != nil {
-		if fields == nil {
-			fields = mapstr.M{}
-		}
+	} else if len(text) != 0 {
```

Comment:
```suggestion
	} else if text != "" {
```

Reply:
Please make this change.

Reply:
Can you extend some of the background for this request?
The linked post is from 2010. Since then the Go compiler got improved.

https://godbolt.org/z/z7ToEfbnT

Reply:
The particular comment that I linked doesn't appeal to performance, but rather clarity. The test here is against the zero value of a string (previously an incorrect nil check); the zero value of a string is `""`. A test for `v != ""` is clearer than `len(v) != 0`, partly by convention and partly by weight. It also is a type-safe comparison that will find changes in types (less of a concern here).

Reply:
The linked comment points me to this statement from Russ Cox from Apr 8, 2010, 8:50:44 PM:
```
It's reasonable to assume that a mature compiler will compile
len(s) == 0 and s == "" into the same, efficient code.
Right now 6g etc do compile s == "" into a function call
while len(s) == 0 is not, but that's been on my to-do list to fix.
```
An additional function call to check something does have a direct performance impact. But for more recent compilers this statement is no longer correct - as shown in https://godbolt.org/z/z7ToEfbnT.

The rest of his comment is:
```
[..] If I'm about to look at element x I typically write
len(s) > x, even for x == 0, but if I care about
"is it this specific string" I tend to write s == "".
```
To me this sounds like a personal preference. Like you said, type-safety is not a concern here.

If the argument is about code style, then I agree to some degree. But I'm wondering if this PR is the right place to start this discussion.
Just looking for the variable `line` (which is usually of type `string`) reveals, that the kind of check I used here is [quite common](https://sourcegraph.com/search?q=context:global+lang:Go+-file:vendor/+repo:github.com/elastic+len%28line%29+%3D%3D+0&patternType=structural&case=yes) across elastic repositories. Code bases evolve and so do programming languages. 

Is this a hard blocker?

Reply:
I'm not going to die on this hill. Please take a look at the strings and bytes packages though for idiomatic tests for string zeroness.

Reply:
I agree with @efd6 about `text != ""` being more readable. And if I understood the posts/comments correctly, there is no real performance implication.

Either way, I don't see it as a blocker.